src/compiler/checker.ts

```diff
@@ -5708,77 +5735,10 @@ namespace ts {
             }
 
             /**
-             * See signatureAssignableTo, signatureAssignableTo
+             * See signatureAssignableTo, compareSignaturesIdentical
              */
             function signatureRelatedTo(source: Signature, target: Signature, reportErrors: boolean): Ternary {

```

Comment:
Can you move all three pieces of code together?
Also, wasn't there a third nearly identical piece of code?